pkg/controller/sub_controller/disaggregated_cluster/computegroups/controller.go

```diff
@@ -259,79 +259,219 @@ func (dcgs *DisaggregatedComputeGroupsController) validateRegex(cgs []dv1.Comput
+	//	if err := k8s.DeleteStatefulset(ctx, dcgs.K8sclient, ddc.Namespace, cgs.StatefulsetName); err != nil {
+	//		cleared = false
+	//		klog.Errorf("disaggregatedComputeGroupsController delete statefulset namespace %s name %s failed, err=%s", ddc.Namespace, cgs.StatefulsetName, err.Error())
+	//		dcgs.K8srecorder.Event(ddc, string(sc.EventWarning), string(sc.CGStatefulsetDeleteFailed), err.Error())
+	//	}
+	//
+	//	if err := k8s.DeleteService(ctx, dcgs.K8sclient, ddc.Namespace, cgs.ServiceName); err != nil {
+	//		cleared = false
+	//		klog.Errorf("disaggregatedComputeGroupsController delete service namespace %s name %s failed, err=%s", ddc.Namespace, cgs.ServiceName, err.Error())
+	//		dcgs.K8srecorder.Event(ddc, string(sc.EventWarning), string(sc.CGServiceDeleteFailed), err.Error())
+	//	}
+	//	if !cleared {
+	//		eCGs = append(eCGs, clearCGs[i])
+	//		continue
+	//	}
+	//	// drop compute group
+	//	cgName := strings.ReplaceAll(cgs.UniqueId, "_", "-")
+	//	cgKeepAmount := int32(0)
+	//	err = dcgs.scaledOutBENodesByDrop(sqlClient, cgName, cgKeepAmount)
+	//	if err != nil {
+	//		klog.Errorf("computeGroupSync ClearResources dropCGBySQLClient failed: %s", err.Error())
+	//		dcgs.K8srecorder.Event(ddc, string(sc.EventWarning), string(sc.CGSqlExecFailed), "computeGroupSync dropCGBySQLClient failed: "+err.Error())
+	//	}
+	//
+	//}
+	//
+	//for i := range eCGs {
+	//	err := dcgs.ClearStatefulsetUnusedPVCs(ctx, ddc, eCGs[i])
+	//	if err != nil {
+	//		klog.Errorf("disaggregatedComputeGroupsController ClearStatefulsetUnusedPVCs clear whole ComputeGroup PVC failed, err=%s", err.Error())
+	//	}
+	//}
+	//for i := range clearCGs {
+	//	err := dcgs.ClearStatefulsetUnusedPVCs(ctx, ddc, clearCGs[i])
+	//	if err != nil {
+	//		klog.Errorf("disaggregatedComputeGroupsController ClearStatefulsetUnusedPVCs clear part ComputeGroup PVC failed, err=%s", err.Error())
+	//	}
+	//}
+	//
+	//ddc.Status.ComputeGroupStatuses = eCGs
+	//
+	//return true, nil
+}
 
-		clearCGs = append(clearCGs, ddc.Status.ComputeGroupStatuses[i])
-		// no need clear should not append.
-	NoNeedAppend:
+func (dcgs *DisaggregatedComputeGroupsController) clearStatefulsets(ctx context.Context, stsNames []string, ddc *dv1.DorisDisaggregatedCluster) error {
+	for _, name := range stsNames {
+		if err := k8s.DeleteStatefulset(ctx, dcgs.K8sclient, ddc.Namespace, name); err != nil {
+			klog.Errorf("DisaggregatedComputeGroupsController clear statefulset failed, namespace=%s, name =%s, err=%s", ddc.Namespace, name, err.Error())
+			return err
+		}
 	}
+	return nil
+}
+
+func (dcgs *DisaggregatedComputeGroupsController) clearSvcs(ctx context.Context, svcNames []string, ddc *dv1.DorisDisaggregatedCluster) error {
+	for _, name := range svcNames {
+		if err := k8s.DeleteService(ctx, dcgs.K8sclient, ddc.Namespace, name); err != nil {
+			klog.Errorf("DisaggregatedComputeGroupsController clear service failed, namespace=%s, name =%s, err=%s", ddc.Namespace, name, err.Error())
+			return err
```

Comment:
It would be better to skip this error and continue deleting, rather than return to end